core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -1106,8 +1106,9 @@ private[chisel3] object Builder extends LazyLogging {
       def foldLayers(l: layer.Layer): Layer = {
         val children = layerAdjacencyList(l)
         val convention = l.convention match {
-          case layer.Convention.Bind => LayerConvention.Bind
-          case _                     => ???
+          case layer.Convention.Bind   => LayerConvention.Bind
+          case layer.Convention.Inline => LayerConvention.Inline
+          case _                       => ???
```

Comment:
I know this is some old code I wrote. In a follow-up, removing the `???` and just letting this error with a default match error may be better.

Reply:
Replaced with covering all cases (add a new convention that isn't handled? -> compile error) and throwing an exception -- wasn't sure if this was the best error-reporting mechanism here or not?  Still want this to evaluate to `Convention`-type value on the non-error path.

(And untested since this isn't exposed outside that package, so should be unreachable...)

Reply:
This approach LGTM.